### PR TITLE
fix(perf): re-introduce js-libp2p perf + add encryption

### PR DESCRIPTION
**Pull request by maschad**
Closes libp2p#291

Also adds noise encryption

Related libp2p/js-libp2p#1991

### CODE REVIEW

**Comment by maschad** [comment]
I've ran this with local infra successfully a few times @mxinden @p-shahi would you be able to trigger a perf run for this on ci?

**Comment by p-shahi** [comment]
Triggered run: https://github.com/libp2p/test-plans/actions/runs/6115444274

**Comment by p-shahi** [comment]
> https://github.com/libp2p/test-plans/actions/runs/6115444274

@maschad the perf run is continuously printing
./impl/js-libp2p/v0.46/perf: line 6: kill: (65610) - No such process

File: perf/runner/terraform.tfstate
Comment (mxinden) [inline perf/runner/terraform.tfstate:1]: Why is this file needed?

File: perf/terraform/configs/local/.terraform.lock.hcl
Comment (mxinden) [inline perf/terraform/configs/local/.terraform.lock.hcl:1]: Please revert this change. In case this change is needed, it should happen in a separate pull request.

**Review by mxinden — COMMENTED** [review]
Thank for the work!

**Comment by mxinden** [comment]
> @maschad the perf run is continuously printing ./impl/js-libp2p/v0.46/perf: line 6: kill: (65610) - No such process

That should be fine as long as the server is able to bind to the provided port and the client is able to connect to it.

File: perf/runner/terraform.tfstate
Comment (maschad) [reply perf/runner/terraform.tfstate:1]: I think this was unintentionally generated.

**Comment by mxinden** [comment]
Thank you for the follow-ups. Triggered a new workflow:

https://github.com/libp2p/test-plans/actions/runs/6187524479/job/16797554461

**Review by mxinden — APPROVED** [review]
